.github/workflows/dependency-update.yml

```
@@ -0,0 +1,175 @@
+
+      - name: Upgrade Playwright (python) and install browsers + OS deps
+        if: ${{ steps.cpr.outputs.pull-request-number != '' }}
+        env:
+          DEBIAN_FRONTEND: noninteractive
+        run: |
+          set -e
+          source .venv/bin/activate
+          python -m pip install --upgrade playwright
+          python -m playwright install --with-deps
+
+      - name: Percy Web run
+        if: ${{ steps.cpr.outputs.pull-request-number != '' }}
+        id: percy_web
+        env:
+          PERCY_TOKEN: ${{ secrets.PERCY_TOKEN_WEB }}
+        run: |
+          set -o pipefail
+          if [ -z "$PERCY_TOKEN" ]; then
+            echo "Missing PERCY_TOKEN_WEB secret; cannot run Percy Web."
+            echo "skipped=1" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+          ( source .venv/bin/activate && npx percy exec -- .venv/bin/python tests/web/test.py ) 2>&1 | tee percy_web.log
+          URL=$(grep -Eo 'https://percy.io[^ ]+' percy_web.log | tail -n1 || true)
+          ID=$(grep -Eo 'Finalized build #[0-9]+' percy_web.log | grep -Eo '[0-9]+' | tail -n1 || true)
+          STATUS=$?
+          # If command returned non-zero, mark failure
+          if [ $STATUS -ne 0 ]; then
+            echo "success=0" >> $GITHUB_OUTPUT
+          else
+            # Percy may still finalize with errors; treat URL presence as success signal
+            if [ -n "$URL" ]; then echo "success=1" >> $GITHUB_OUTPUT; else echo "success=0" >> $GITHUB_OUTPUT; fi
+          fi
+          echo "url=${URL}" >> $GITHUB_OUTPUT
+          echo "id=${ID}" >> $GITHUB_OUTPUT
+
+      - name: Comment on PR with Percy result
+        if: ${{ steps.cpr.outputs.pull-request-number != '' }}
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const prNumber = Number('${{ steps.cpr.outputs.pull-request-number }}');
+            const url = '${{ steps.percy_web.outputs.url }}';
+            const id = '${{ steps.percy_web.outputs.id }}';
+            const skipped = '${{ steps.percy_web.outputs.skipped }}' === '1';
+            const success = '${{ steps.percy_web.outputs.success }}' === '1';
+
+            let body;
+            if (skipped) {
+              body = `## Percy Web\nSkipped (missing \`PERCY_TOKEN_WEB\`).`;
+            } else if (success && url) {
+              body = `## Percy Web\n✅ Build #${id}\n${url}`;
+            } else {
+              body = `## Percy Web\n❌ Build failed (see logs in workflow artifacts).`;
+              if (url) body += `\nFinalized build URL (may contain error details):\n${url}`;
+            }
+
+            await github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: prNumber,
+              body,
+            });
+
+      - name: Upload Percy logs
+        if: always() && ${{ steps.cpr.outputs.pull-request-number != '' }}
```

```suggestion
        if: always() && steps.cpr.outputs.pull-request-number != ''
```

The statement inside of if is always treated as expression -- Check if we can use ^

---

resolved